pgvectorscale/src/access_method/build.rs

```diff
@@ -91,6 +91,10 @@ pub extern "C" fn ambuild(
         DistanceType::from_u16(result)
     };
 
+    if distance_type == DistanceType::InnerProduct && opt.get_storage_type() == StorageType::Plain {
```

Comment:
This seems a bit brittle. Can we ensure the only storage type allowed with inner product is SBQ instead?